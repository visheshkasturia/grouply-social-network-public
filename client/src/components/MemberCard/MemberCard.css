.member-card-container {
  display: grid;
  grid-template-columns: 10% 30% 30% 30%;
  grid-template-rows: 80% 20%;
  height: 110px;
  border: solid 3px black;
  border-radius: 10px;
  margin: 15px;
}

.member-card-SNo {
  grid-column: 1/ span 1;
  grid-row: 1/ span 1;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 24px;
  font-weight: 400;
}

.member-card-name {
  grid-column: 2/ span 1;
  grid-row: 1/ span 1;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 24px;
  font-weight: 400;
}

.member-card-addAdmin {
  grid-column: 3/ span 1;
  grid-row: 1/ span 1;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 24px;
  font-weight: 400;
}

.member-card-removeMember {
  grid-column: 4/ span 1;
  grid-row: 1/ span 1;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 24px;
  font-weight: 400;
}

.add-admin-status {
  grid-column: 1/ span 4;
  grid-row: 2/ span 1;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 16px;
  font-weight: 400;
}