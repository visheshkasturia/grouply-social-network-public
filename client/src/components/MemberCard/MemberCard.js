import React, { useState } from 'react';
import './MemberCard.css';
import { removeMemberFromGroup } from '../../apiCalls/removeMemberFromGroup';
import { makeAdmin } from '../../apiCalls/makeAdmin';
/* eslint-disable */

function MemberCard({ member, user, i, getChangedMemberList, group }) {
  const [status, setStatus] = useState();

  const addAdmin = () => {
    makeAdmin(user.userName, member, group._id).then ((data) => {
      if (data != 'Unauthorized access'){
        setStatus(data)
      }
    })
    return
  }

  const removeMember = () => {
    removeMemberFromGroup(member, user._id, group._id)
      .then((data) => {
        if (data != 'Unauthorized access'){
          getChangedMemberList(data.allMembers);
        }
      })
      .catch((err) => {
        console.log(err)
      });
    return
  }

  return (
    <div className="member-card-container">
      <div className="member-card-SNo flex-center"> {i} </div>
      <div className="member-card-name flex-center"> {member} </div>
      <div className="member-card-addAdmin flex-center" >
      <button type="button" onClick={addAdmin} >Make Admin</button>
      </div>
      <div className="member-card-removeMember flex-center">
      <button type="button" onClick={removeMember} >Remove Member</button>
      </div>
      <div className="add-admin-status">{status}</div>
    </div>
  )
}

export default MemberCard;
