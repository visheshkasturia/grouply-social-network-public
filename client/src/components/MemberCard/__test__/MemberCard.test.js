import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import MemberCard from '../MemberCard'

const admin = 'abc'
const member = 'test1'
const post = '1'
const tag = 'tag'
const group = {
  groupName: "test",
  description: "desc",
  admins: [admin],
  allMembers: [member],
  allPosts: [post],
  tags: [tag]
}

it('MemberCard renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MemberCard member={member} group={group} />, div);
});

it('MemberCard matches snapshot', () => {
  const tree = renderer.create(<MemberCard member={member} group={group} />).toJSON();
  expect(tree).toMatchSnapshot();
});
