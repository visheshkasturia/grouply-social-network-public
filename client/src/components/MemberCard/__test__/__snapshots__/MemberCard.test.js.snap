// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberCard matches snapshot 1`] = `
<div
  className="member-card-container"
>
  <div
    className="member-card-SNo flex-center"
  >
     
     
  </div>
  <div
    className="member-card-name flex-center"
  >
     
    test1
     
  </div>
  <div
    className="member-card-addAdmin flex-center"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Make Admin
    </button>
  </div>
  <div
    className="member-card-removeMember flex-center"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Remove Member
    </button>
  </div>
  <div
    className="add-admin-status"
  />
</div>
`;
